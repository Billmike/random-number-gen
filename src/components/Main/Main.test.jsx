import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Main } from './index';

jest.mock('file-saver', () => ({
  saveAs: jest.fn(),
}));

test('it should mount without crashing', () => {
  const props = {
    numbers: []
  }
  const component = renderer.create(<Main {...props} />);
  let test = component.toJSON();

  expect(test).toMatchSnapshot()
})

test('should verify the addPhoneNumber method handler is called correctly', () => {
  const props = {
    numbers: [],
    addNumbers: jest.fn()
  };
  const wrapper = shallow(<Main {...props} />);
  wrapper.setState({ generatingValue: 20 })
  const instance = wrapper.instance();
  instance.onAddPhoneNumber();

  expect(props.addNumbers).toHaveBeenCalled()
});

test('should verify the numbers are sorted in ascending order', () => {
  const props = {
    numbers: [],
    sortNumbersAscending: jest.fn(),
    sortNumbersDescending: jest.fn()
  };
  const wrapper = shallow(<Main {...props} />);
  const event = {
    target: {
      value: "Ascending"
    }
  };
  const instance = wrapper.instance();
  instance.onSortNumber(event);

  expect(props.sortNumbersAscending).toHaveBeenCalled()
})

test('should verify the numbers are sorted in descending order', () => {
  const props = {
    numbers: [],
    sortNumbersAscending: jest.fn(),
    sortNumbersDescending: jest.fn()
  };
  const wrapper = shallow(<Main {...props} />);
  const event = {
    target: {
      value: "Descending"
    }
  };
  const instance = wrapper.instance();
  instance.onSortNumber(event);

  expect(props.sortNumbersDescending).toHaveBeenCalled()
});

test('should verify the onGeneratingValue method is called and state is updated', () => {
  const props = {
    numbers: []
  };
  const wrapper = shallow(<Main {...props} />);
  const event = {
    target: {
      value: 20
    }
  };
  wrapper.setState({ generatingValue: '', hasGeneratedNumbers: false });
  const instance = wrapper.instance();
  instance.onGeneratingValueChange(event);

  expect(wrapper.state().generatingValue).toEqual(20);
  expect(wrapper.state().hasGeneratedNumbers).toBe(true);
})

test('should verify the onDownloadPhoneNumbers method is called correctly', () => {
  const props = {
    numbers: [
      {
        userPhoneNumber: 2345678
      },
      {
        userPhoneNumber: 8494343
      }
    ]
  }
  const wrapper = shallow(<Main {...props} />);
  const instance = wrapper.instance();

  instance.onDownloadPhoneNumbers();

  expect(wrapper).toMatchSnapshot()
})

