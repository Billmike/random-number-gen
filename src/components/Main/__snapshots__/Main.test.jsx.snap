// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should mount without crashing 1`] = `
<div
  className="container"
>
  <h4
    className="heading-text"
  >
    Random number generator
  </h4>
  <div
    className="button-container"
  >
    <input
      className="generating-value"
      onChange={[Function]}
      type="number"
      value=""
    />
    <div>
      <button
        className="button-style"
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "grey",
            "cursor": "not-allowed",
          }
        }
      >
        Generate Phone Number
      </button>
      <button
        className="download-button"
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "grey",
            "cursor": "not-allowed",
          }
        }
      >
        Download Numbers
      </button>
    </div>
  </div>
  <div
    className="container-style"
  />
</div>
`;

exports[`should verify the onDownloadPhoneNumbers method is called correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    numbers={
      Array [
        Object {
          "userPhoneNumber": 2345678,
        },
        Object {
          "userPhoneNumber": 8494343,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="heading-text"
        >
          Random number generator
        </h4>,
        <div
          className="button-container"
        >
          <input
            className="generating-value"
            onChange={[Function]}
            type="number"
            value=""
          />
          <div>
            <GeneratePhoneNumberButton
              generatingValue=""
              onAddPhoneNumber={[Function]}
            />
            <DownloadButton
              numbers={
                Array [
                  Object {
                    "userPhoneNumber": 2345678,
                  },
                  Object {
                    "userPhoneNumber": 8494343,
                  },
                ]
              }
              onDownloadPhoneNumbers={[Function]}
            />
          </div>
          <div
            className="select-container"
          >
            <label
              className="select-label"
            >
              Choose sorting order
            </label>
            <select
              onChange={[Function]}
            >
              <option
                value="Ascending"
              >
                Ascending
              </option>
              <option
                value="Descending"
              >
                Descending
              </option>
            </select>
          </div>
        </div>,
        false,
        <DisplayNumber
          hasGeneratedNumbers={false}
          phoneNumbers={
            Array [
              Object {
                "userPhoneNumber": 2345678,
              },
              Object {
                "userPhoneNumber": 8494343,
              },
            ]
          }
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Random number generator",
          "className": "heading-text",
        },
        "ref": null,
        "rendered": "Random number generator",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="generating-value"
              onChange={[Function]}
              type="number"
              value=""
            />,
            <div>
              <GeneratePhoneNumberButton
                generatingValue=""
                onAddPhoneNumber={[Function]}
              />
              <DownloadButton
                numbers={
                  Array [
                    Object {
                      "userPhoneNumber": 2345678,
                    },
                    Object {
                      "userPhoneNumber": 8494343,
                    },
                  ]
                }
                onDownloadPhoneNumbers={[Function]}
              />
            </div>,
            <div
              className="select-container"
            >
              <label
                className="select-label"
              >
                Choose sorting order
              </label>
              <select
                onChange={[Function]}
              >
                <option
                  value="Ascending"
                >
                  Ascending
                </option>
                <option
                  value="Descending"
                >
                  Descending
                </option>
              </select>
            </div>,
          ],
          "className": "button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "generating-value",
              "onChange": [Function],
              "type": "number",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <GeneratePhoneNumberButton
                  generatingValue=""
                  onAddPhoneNumber={[Function]}
                />,
                <DownloadButton
                  numbers={
                    Array [
                      Object {
                        "userPhoneNumber": 2345678,
                      },
                      Object {
                        "userPhoneNumber": 8494343,
                      },
                    ]
                  }
                  onDownloadPhoneNumbers={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "generatingValue": "",
                  "onAddPhoneNumber": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "numbers": Array [
                    Object {
                      "userPhoneNumber": 2345678,
                    },
                    Object {
                      "userPhoneNumber": 8494343,
                    },
                  ],
                  "onDownloadPhoneNumbers": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  className="select-label"
                >
                  Choose sorting order
                </label>,
                <select
                  onChange={[Function]}
                >
                  <option
                    value="Ascending"
                  >
                    Ascending
                  </option>
                  <option
                    value="Descending"
                  >
                    Descending
                  </option>
                </select>,
              ],
              "className": "select-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Choose sorting order",
                  "className": "select-label",
                },
                "ref": null,
                "rendered": "Choose sorting order",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value="Ascending"
                    >
                      Ascending
                    </option>,
                    <option
                      value="Descending"
                    >
                      Descending
                    </option>,
                  ],
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Ascending",
                      "value": "Ascending",
                    },
                    "ref": null,
                    "rendered": "Ascending",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Descending",
                      "value": "Descending",
                    },
                    "ref": null,
                    "rendered": "Descending",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "hasGeneratedNumbers": false,
          "phoneNumbers": Array [
            Object {
              "userPhoneNumber": 2345678,
            },
            Object {
              "userPhoneNumber": 8494343,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="heading-text"
          >
            Random number generator
          </h4>,
          <div
            className="button-container"
          >
            <input
              className="generating-value"
              onChange={[Function]}
              type="number"
              value=""
            />
            <div>
              <GeneratePhoneNumberButton
                generatingValue=""
                onAddPhoneNumber={[Function]}
              />
              <DownloadButton
                numbers={
                  Array [
                    Object {
                      "userPhoneNumber": 2345678,
                    },
                    Object {
                      "userPhoneNumber": 8494343,
                    },
                  ]
                }
                onDownloadPhoneNumbers={[Function]}
              />
            </div>
            <div
              className="select-container"
            >
              <label
                className="select-label"
              >
                Choose sorting order
              </label>
              <select
                onChange={[Function]}
              >
                <option
                  value="Ascending"
                >
                  Ascending
                </option>
                <option
                  value="Descending"
                >
                  Descending
                </option>
              </select>
            </div>
          </div>,
          false,
          <DisplayNumber
            hasGeneratedNumbers={false}
            phoneNumbers={
              Array [
                Object {
                  "userPhoneNumber": 2345678,
                },
                Object {
                  "userPhoneNumber": 8494343,
                },
              ]
            }
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Random number generator",
            "className": "heading-text",
          },
          "ref": null,
          "rendered": "Random number generator",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="generating-value"
                onChange={[Function]}
                type="number"
                value=""
              />,
              <div>
                <GeneratePhoneNumberButton
                  generatingValue=""
                  onAddPhoneNumber={[Function]}
                />
                <DownloadButton
                  numbers={
                    Array [
                      Object {
                        "userPhoneNumber": 2345678,
                      },
                      Object {
                        "userPhoneNumber": 8494343,
                      },
                    ]
                  }
                  onDownloadPhoneNumbers={[Function]}
                />
              </div>,
              <div
                className="select-container"
              >
                <label
                  className="select-label"
                >
                  Choose sorting order
                </label>
                <select
                  onChange={[Function]}
                >
                  <option
                    value="Ascending"
                  >
                    Ascending
                  </option>
                  <option
                    value="Descending"
                  >
                    Descending
                  </option>
                </select>
              </div>,
            ],
            "className": "button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "generating-value",
                "onChange": [Function],
                "type": "number",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <GeneratePhoneNumberButton
                    generatingValue=""
                    onAddPhoneNumber={[Function]}
                  />,
                  <DownloadButton
                    numbers={
                      Array [
                        Object {
                          "userPhoneNumber": 2345678,
                        },
                        Object {
                          "userPhoneNumber": 8494343,
                        },
                      ]
                    }
                    onDownloadPhoneNumbers={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "generatingValue": "",
                    "onAddPhoneNumber": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "numbers": Array [
                      Object {
                        "userPhoneNumber": 2345678,
                      },
                      Object {
                        "userPhoneNumber": 8494343,
                      },
                    ],
                    "onDownloadPhoneNumbers": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    className="select-label"
                  >
                    Choose sorting order
                  </label>,
                  <select
                    onChange={[Function]}
                  >
                    <option
                      value="Ascending"
                    >
                      Ascending
                    </option>
                    <option
                      value="Descending"
                    >
                      Descending
                    </option>
                  </select>,
                ],
                "className": "select-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Choose sorting order",
                    "className": "select-label",
                  },
                  "ref": null,
                  "rendered": "Choose sorting order",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <option
                        value="Ascending"
                      >
                        Ascending
                      </option>,
                      <option
                        value="Descending"
                      >
                        Descending
                      </option>,
                    ],
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Ascending",
                        "value": "Ascending",
                      },
                      "ref": null,
                      "rendered": "Ascending",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Descending",
                        "value": "Descending",
                      },
                      "ref": null,
                      "rendered": "Descending",
                      "type": "option",
                    },
                  ],
                  "type": "select",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "hasGeneratedNumbers": false,
            "phoneNumbers": Array [
              Object {
                "userPhoneNumber": 2345678,
              },
              Object {
                "userPhoneNumber": 8494343,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
